### DISCUSSION TITLE
Vicuna v1.5 released with 4K and 16K context lengths

### DISCUSSION THREAD

**Discussion by ianscrivener** [ggml-org/ggml#431] [General]
"Vicuna v1.5 series, featuring 4K and 16K context lengths with improved performance on almost all benchmarks.. based on the commercial-friendly Llama2"

![image](https://github.com/ggerganov/ggml/assets/163229/a4ab22f0-3f5e-44e2-957b-5bda2cdf3be3)

Excited to see llama.cpp support soon?!

**Comment by klosax** [comment]
The 16k model should already work using the parameter `--rope-freq-scale 4.0`

**Reply by ianscrivener** [reply]
hmm.. no luck for me on Mac with Metal.
Just starts outputting unpromtped... mixed languages and a lots of new lines

`./main -t 4 -ngl 1 --model vicuna-7b-v1.5-16k.ggmlv3.q4_0.bin  --ctx-size 2048 --rope-freq-scale 4.0 -i`

`./main -t 4 -ngl 1 --model vicuna-7b-v1.5-16k.ggmlv3.q4_0.bin  --ctx-size 16384 --rope-freq-scale 4.0 -i`

**Reply by ianscrivener** [reply]
am I missing something?

**Reply by klosax** [reply]
I got it wrong. The `--rope-freq-scale` should be the inverse of the context scaling factor.

In PR https://github.com/ggerganov/llama.cpp/pull/2544 I added a context scaling parameter ` --rope-scale` which makes more sense:
`--rope-scale 4.0`
